### DISCUSSION TITLE
Easy way to fix a messed up library?

### DISCUSSION THREAD

**Discussion by GoodClover** [beetbox/beets#3940] [Q&A]
Hello, I learnt of beets a few days ago and absolutely love it.
However today I managed to mess my library up, `-i` didn't work (as I expected) and I ended up having my entire thing re-done, so non-album tracks were attempted to be moved into albums.
After trying to fix it, I now have everything in a way where everything has been imported as a singleton, so my albums are split up.

Is there an easy way to re-group albums?

Also, how can I re-run the auto-tagger on already imported music?

**Comment by sampsyo** [comment]
Hmm… what do your files currently look like? If the files are sorted into albums correctly, you could try a non-autotagged import (`beet import -A`). If they're scattered around haphazardly, you might try the `--group-albums` flag… but use this one with care and backups; it can seem unpredictable.

You can do a "re-import" by just importing files that are already in your library, or using the `-L` flag to re-import music based on a query.

**Reply by GoodClover** [reply]
Hmm not sure what `-g`/`--group-albums` is intended to do, but it doesn't appear to have changed anything.

At current I'm able to get things organised in a way where either everything is a "singleton" or where the albums are all correct, but things that should be singletons are just laid around the place.

**Reply by wisp3rwind** [reply]
> or where the albums are all correct, but things that should be singletons are just laid around the place.

That does sound roughly like what you want to achieve, right? At least albums and singletons seem to be properly assigned? Please be more specific on what is wrong with the singletons "just laid around the place". Are they simply not being moved to the locations you'd like to?
